docs/generate_mermaid_flowcharts.py

```diff
@@ -0,0 +1,417 @@
+        :return: Display name for the ResourceNode.
+        """
+        if bind_kwargs is None:
+            res_display = self.res_name
+        else:
+            bind_kwargs = self.remove_excluded_parameters(bind_kwargs)
+            res_display = ",\n".join("{}={}".format(k, v) for k, v in bind_kwargs)
+
+            if len(bind_kwargs) > 1:
+                res_display = "\n" + res_display
+
+        return f"{self.res_name}({res_display})"
+
+    def get_resource_id(self, bind_kwargs=None):
+        """
+        Get the ID for the ResourceNode.
+
+        :param bind_kwargs: Optional resource function kwargs to run the function with.
+        :return: ID for the ResourceNode.
+        """
+        if bind_kwargs is None:
+            return self.res_name
+        else:
+            res_arg_str = "_".join(
+                "{}_{}".format(k, v)
+                for k, v in self.remove_excluded_parameters(bind_kwargs)
+            )
+            return f"{self.res_name}_{res_arg_str}"
+
+    def add_resource_to_flowchart(self, flowchart, resource, bind_kwargs=None):
+        """
+        Run a resource function and add its resource nodes to the flowchart.
+
+        :param flowchart: QCFlowchart object.
+        """
+        if hasattr(resource, "versions"):
+            paths = [r.path for r in resource.versions.values()]
+        elif hasattr(resource, "path"):
+            paths = [resource.path]
+        elif isinstance(resource, str):
+            paths = [resource]
+        else:
+            err_msg = "" if bind_kwargs is None else f"with {bind_kwargs}"
+            raise ValueError(
+                f"Resource {self.res_obj} {err_msg}did not return a valid resource."
+            )
+
+        for p in paths:
+            if check_file_exists_raise_error(p, error_if_exists=False):
+                flowchart.add_node(
+                    p,
+                    type="resource",
+                    color=NODE_COLORS["resource"],
+                    label=self.get_resource_display_name(bind_kwargs),
+                    id=self.get_resource_id(bind_kwargs),
+                    href=get_function_doc_link(self.module, self.res_name),
+                    res_node=self,
+                )
+
+    def add_to_flowchart(self, flowchart):
+        """
+        Run a resource function and add its resource nodes to the flowchart.
+
+        :param flowchart: QCFlowchart object.
+        """
+        if self.is_module_func:
+            res_sig = self.get_signature()
+            for bind_kwargs in self.get_resource_kwargs_combos():
+                res_bind = res_sig.bind(**dict(bind_kwargs))
+
+                try:
+                    self.add_resource_to_flowchart(
+                        flowchart,
+                        self.res_obj(*res_bind.args, **res_bind.kwargs),
+                        bind_kwargs=bind_kwargs,
+                    )
+                except (ValueError, DataException, KeyError):
+                    continue
```

Comment:
should this have a warning message (since it's currently a silent failure)? or are you expecting that some resources will fail